File: DailyChallenges/Scala/LeetCode_1572.scala
object Solution {
    def diagonalSum(mat: Array[Array[Int]]): Int = {
        val n = mat.length
        val p = mat.foldLeft((0, 0)){(p, a) => (p._1 + a(p._2) + a(n-1-p._2), p._2+1)}
        if (n%2 == 1) p._1 - mat(n/2)(n/2) else p._1
    }
}

File: DailyChallenges/CPP/LeetCode_1572.cpp
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int n = mat.size(), ans = 0;
        for (int i=0; i< n; ++i){
            ans += mat[i][i];
            ans += mat[i][n-i-1];
        }
        if (n%2 == 1) ans -= mat[n/2][n/2];
        return ans;
    }
};
